plugins/nordic-hid/fu-nordic-hid-cfg-channel.c

```diff
@@ -541,6 +544,65 @@ fu_nordic_hid_cfg_channel_get_bl_name(FuNordicHidCfgChannel *self, GError **erro
 	return TRUE;
 }
 
+static gboolean
+fu_nordic_hid_cfg_channel_get_devinfo(FuNordicHidCfgChannel *self, GError **error)
+{
+	guint8 event_id = 0;
+	g_autoptr(FuNordicCfgChannelMsg) res = g_new0(FuNordicCfgChannelMsg, 1);
+
+	/* query for the devinfo if the board supports it */
+	if (fu_nordic_hid_cfg_channel_get_event_id(self, "dfu", "devinfo", &event_id)) {
```

Comment:
the other way around perhaps? then you can return on failure.

Reply:
Support for `devinfo` command handling is optional, but if a given board does not support it, the generation parameter must be defined in the quirk file (handled in similar way as for the bootloader). We cannot return here on failure, because then we would bypass check that generation is set.
Or you meant something else?